recipe/0001-Use-python_extension_module.patch

```diff
@@ -0,0 +1,57 @@
+--- slycot/CMakeLists.txt
++++ slycot/CMakeLists.txt
+@@ -112,8 +112,7 @@
+   transform.py ${CMAKE_CURRENT_BINARY_DIR}/version.py)
+ 
+ set(SLYCOT_MODULE "_wrapper")
+-set(GENERATED_MODULE
+-  ${CMAKE_CURRENT_BINARY_DIR}/${SLYCOT_MODULE}${PYTHON_EXTENSION_MODULE_SUFFIX})
++find_package(PythonExtensions REQUIRED)
+ 
+ 
+ set(CMAKE_Fortran_FLAGS )
+@@ -126,33 +125,18 @@
+   )
+ 
+ add_library(
+-  ${SLYCOT_MODULE} SHARED
++  ${SLYCOT_MODULE} MODULE
+   SLYCOTmodule.c _wrappermodule.c _wrapper-f2pywrappers.f
+   "${PYTHON_SITE}/numpy/f2py/src/fortranobject.c"
+   ${FSOURCES})
+ 
+-set(CMAKE_SHARED_LIBRARY_PREFIX "")
+-if (WIN32)
+-  set(CMAKE_SHARED_LIBRARY_SUFFIX ".pyd")
+-endif()
+-set_target_properties(${SLYCOT_MODULE} PROPERTIES
+-  OUTPUT_NAME "_wrapper")
+-if (WIN32)
+-  target_link_libraries(${SLYCOT_MODULE} PUBLIC
+-    ${PYTHON_LIBRARIES} ${LAPACK_LIBRARIES}) 
+-endif()
++target_link_libraries(${SLYCOT_MODULE}
```

Comment:
@repagh, I had to remove the `PUBLIC` part because `python_extension_module` invokes `target_link_libraries` without it:
https://github.com/scikit-build/scikit-build/blob/0.10.0/skbuild/resources/cmake/FindPythonExtensions.cmake#L423
https://github.com/scikit-build/scikit-build/blob/0.10.0/skbuild/resources/cmake/targetLinkLibrariesWithDynamicLookup.cmake#L579
and CMake does not like to mix and match those different uses.

I'm very much no CMake expert so can't tell if that'll have any implications for you.
Other than that, if the patch makes sense to you, feel free to incorporate it upstream.